src/test/java/com/skalashynski/spring/springboot/service/StudentServiceTest.java

```diff
@@ -0,0 +1,231 @@
+        given(studentRepository.findAll()).willReturn(expected);
+
+        List<Student> actual = studentService.getAll();
+
+        then(studentRepository)
+                .should()
+                .findAll();
+        assertEquals(expected, actual);
+    }
+
+    @ParameterizedTest
+    @ValueSource(longs = {Long.MIN_VALUE, Long.MAX_VALUE, 20L})
+    @DisplayName("Should delete an entity by id and return true if exists")
+    void deleteTest_ok(long id) {
+        boolean expected = provideStudents()
+                .anyMatch(s -> s.getId() == id);
+        given(studentRepository.existsById(id))
+                .willReturn(expected);
+
+        boolean actual = studentService.delete(id);
+
+        then(studentRepository)
+                .should()
+                .deleteById(id);
+        assertTrue(expected);
+        assertTrue(actual);
+    }
+
+    @ParameterizedTest
+    @ValueSource(longs = {1234L, 5L, 100L})
+    @DisplayName("Shouldn't delete any entity and return false if doesn't exist")
+    void deleteTest_fail(long id) {
+        boolean expected = provideStudents()
+                .anyMatch(s -> s.getId() == id);
+        given(studentRepository.existsById(id))
+                .willReturn(expected);
+
+        boolean actual = studentService.delete(id);
+
+        then(studentRepository)
+                .should(never())
+                .deleteById(id);
+        assertFalse(expected);
+        assertFalse(actual);
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideArgsForUpdateTest")
+    @DisplayName("Should update an entity by id and return it if exists")
+    void updateTest_ok(long id, Student student) {
+        Optional<Student> foundById = provideStudents()
+                .filter(s -> s.getId() == id)
+                .findFirst();
+        given(studentRepository.findById(id))
+                .willReturn(foundById);
+
+        Student expected = new Student(id, student.getFirstName(), student.getLastName(),
+                student.getBirthday(), foundById.orElseThrow().getCreatedAt());
+        given(studentRepository.save(expected))
+                .willReturn(expected);
+
+        Student actual = studentService.update(id, student);
+
+        then(studentRepository)
+                .should()
+                .save(expected);
+        assertEquals(expected, actual);
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideArgsForUpdateFailTest")
+    @DisplayName("Should throw the ApiException if entity doesn't exist with given id")
+    void updateTest_fail(long id, Student student) {
+        Optional<Student> foundById = provideStudents()
+                .filter(s -> s.getId() == id)
+                .findFirst();
+        given(studentRepository.findById(id))
+                .willReturn(foundById);
+
+        assertThrows(ApiException.class,
+                () -> studentService.update(id, student),
+                "Can't find student with id: " + id);
+    }
```

Comment:
Create a String constant in the source code for `"Can't find student with id: "`, make it `public static` and use this constant in test and source code.

Reply:
I did it by adding `enum ApiExceptionMessage`.